src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -129,6 +137,38 @@ function createModel(program: Program): any {
     if (authentication) {
         auth = processServiceAuthentication(authentication);
     }
+    const consumes = getConsumes(program, serviceNamespaceType);
+    let contentTypeParameter = undefined;
+    const requestMediaTypes: string[] = [];
+    if (consumes && consumes.length > 0) {
+        contentTypeParameter = {
+            Name: "content_type",
+            NameInRequest: "content_type",
+            Type: {
+                Name: "String",
+                Kind: InputTypeKind.String,
+                IsNullable: false
+            } as InputPrimitiveType,
+            Location: RequestLocation.Header,
+            IsApiVersion: false,
+            IsResourceParameter: false,
+            IsContentType: true,
+            IsRequired: true,
+            IsEndpoint: false,
+            SkipUrlEncoding: false,
+            Explode: false,
+            Kind: InputOperationParameterKind.Method,
+            DefaultValue: {
+                Type: {
+                    Name: "String",
+                    Kind: InputTypeKind.String,
+                    IsNullable: false
+                } as InputPrimitiveType,
+                Value: consumes[0]
```

Comment:
what if Comsumes contains multple values?

Reply:
updated. only when one value, it will set as default value for later usage in codegen. If multiple values, don't set default value.

Reply:
Let's skip this case for now